extensions/widgets/header/header.lcb

```diff
@@ -68,551 +94,1036 @@ Sets the header actions display style of the control.
+      set the paint of this canvas to fetchPaint("searchIcon")
+      fill fetchPath("searchIcon") on this canvas
+   end if
+
 end handler
 
 public handler OnMouseDown()
+   -- the icons change color when they are pressed down
+   variable tSelectedDown as Integer
+   put fetchSelectedItem() into tSelectedDown
 
-	-- the icons change color when they are pressed down
-	variable tSelectedDown as optional Integer
-	put getSelectedIcon() into tSelectedDown
-	
-	if tSelectedDown > 0 then
-		put tSelectedDown into mSelectedIcon
-		redraw all
-	end if
+   if tSelectedDown > 0 then
+      put tSelectedDown into mSelectedAction
+      redraw all
+   end if
 
 end handler
 
 public handler OnMouseUp()
 
-	-- the icons change back to their original colors on mouse up
-	put 0 into mSelectedIcon
-	redraw all
-	
-	-- the relevant commands are dispatched to the script object
-	variable tSelectedUp as optional integer
-	put getSelectedIcon() into tSelectedUp
-	
-	if tSelectedUp > 1 then
-		
-		variable tActionCount as optional integer
-		variable tNumActions as Number
-		variable tX as Number
-		
-		put the number of elements in (element 1 of mActions) into tNumActions
-		put 1 into tActionCount	
-		
-		repeat with tX from 1 up to tNumActions
-		
-			if tActionCount is tSelectedUp-1 then
-				variable tLabel as String
-				put element tX of (element 1 of mActions) into tLabel
-				post "headerAction" with [tLabel]
-			end if
-				
-			add 1 to tActionCount
-		end repeat
-		
-	else if tSelectedUp = 1 then
-		post "backAction"
-	end if
-	
-end handler
-
-private handler getPaint(in pObject as String) as Paint
-
-	if pObject is "background" then
-		return solid paint with color [246/255, 246/255, 247/255]
-		
-	else if pObject is "actions" then
-		return solid paint with  color [51/255, 153/255, 1.0] 
-		
-	else if pObject is "selected action" then
-		return solid paint with color [51/255, 153/255, 1.0, 0.25]
-		
-	else if pObject is "back button" then
-		return solid paint with  color [51/255, 153/255, 1.0] 
-		
-	else if pObject is "back button selected" then
-		return solid paint with color [51/255, 153/255, 1.0, 0.25]
-	
-	else if pObject is "line" then
-		return solid paint with color [178/255, 178/255, 178/255]
-		
-	else if pObject is "title" then
-		 return solid paint with color kBlack
-		
-	end if
-	
+   -- the icons change back to their original colors on mouse up
+   put 0 into mSelectedAction
+   redraw all
+
+   -- the relevant commands are dispatched to the script object
+   variable tSelectedUp as Integer
+   put fetchSelectedItem() into tSelectedUp
+
+   if tSelectedUp > 2 then
+
+      variable tActionCount as Integer
+      variable tNumActions as Integer
+
+      put the number of elements in element 1 of mHeaderActions into tNumActions
+
+      repeat with tActionCount from 1 up to tNumActions
+
+         if tActionCount is tSelectedUp-2 then
+            post "headerAction" with mHeaderActions[tActionCount]["name"]
```

Comment:
The 'with' argument of post has to be a list (even if 1 element), so this should be 

   post "headerAction" with [mHeaderActions[tActionCount]["name"]]